src/registry.rs

```diff
@@ -41,7 +41,21 @@ pub struct Registry {
 	string_table: Interner<&'static str>,
 	#[serde(skip)]
 	type_table: Interner<AnyTypeId>,
-	types: Vec<TypeIdDef>,
+	#[serde(serialize_with = "serialize_registry_types")]
+	types: BTreeMap<UntrackedSymbol<std::any::TypeId>, TypeIdDef>,
+}
+
+/// Serializes the types of the registry by removing their unique IDs
+/// and instead serialize them in order of their removed unique ID.
+fn serialize_registry_types<S>(
+	types: &BTreeMap<UntrackedSymbol<std::any::TypeId>, TypeIdDef>,
```

Comment:
`std::any::TypeId` won't work for wasm / no_std build. `core::any::TypeId` should do?

Reply:
Didn't test this in ink! since there metadata is only generated in `std` environment. Will fix it quickly.